src/app/components/form-fields/date-time-field/date-time-field.component.ts

```diff
@@ -29,6 +41,35 @@ export class DateTimeFieldComponent extends BaseDirectFormField<Date, [DateTimeF
         return true;
     };
 
+    public validTimeChange(event: boolean): void {
+        this.isTimeValid = event;
+        this.onValueChange(event);
+    }
+
+    public onValueChange(event: unknown): void {
+        event;
+        if (this.courseTimeZone === '' || this.courseTimeZone === DateTimeFieldComponent.localTimeZone) {
```

Comment:
Since the comparsion of time zone is done by comparing IANA time zone canonical names, there might be a possibility that the local time zone and the course time zone have different IANA time zone canonical names but the same time value (e.g., `America/Chihuahua` and `America/Guatemala` are both `UTC-6` offset without DST).

In such case, the `Course Time` label will be displayed with the same time as configured by the input field, which could be confusing for the user. To solve this issue, the comparsion could be made between `utcToZonedTime(time, courseTimeZone)` and `utcToZonedTime(time, localTimeZone)` instead.

Reply:
A worry I have here is that if we keep showing the course time preview being the same time as configured in the field, the user might gradually tend to ignore whatever being shown under the date-time field. Currently I think either way would be fine, but future changes might be needed based on user feedback & behaviors.

Reply:
In terms of course timezone mismatch warning, I suggest showing a warning sign below the course name & term name. Clicking the sign will display the modal including details about mismatched timezone.